### PR TITLE
Merging feature branch for Package Deployment service to Main

### PR DESCRIPTION
**Pull request by Shubbhang351**
1. **What this PR does / why we need it**:

Merging feature branch for Package Deployment service to Main

### CODE REVIEW

File: internal/resources/tanzupackageinstall/package_install_resource_test.go [hunk 1]
```diff
@@ -0,0 +1,321 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package tanzupackageinstall
```
Comment (ramya-bangera) [inline internal/resources/tanzupackageinstall/package_install_resource_test.go:6]: @Shubbhang351 - Shobha recently added the build tags to acceptance tests as part adding automation jobs of TMC terraform. Its daily simple task, can you work with Shobha to include build tags for these new resources as part of this change?

**Comment by ramya-bangera** [comment]
Overall LGTM

**Review by ramya-bangera — DISMISSED** [review]

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]

File: internal/resources/package/package_mock_test.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+const (
+	https                = "https:/"
+	clAPIVersionAndGroup = "v1alpha1/clusters"
+	apiSubGroup          = "namespaces"
+	apiKind              = "tanzupackage/metadatas"
+	packages             = "packages"
+)
+
+// nolint: unparam
```
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply internal/resources/package/package_mock_test.go]: updated

File: internal/resources/packages/packages_mock_test.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+const (
+	https                = "https:/"
+	clAPIVersionAndGroup = "v1alpha1/clusters"
+	apiSubGroup          = "namespaces"
+	apiKind              = "tanzupackage/metadatas"
+	packages             = "packages"
+)
+
+// nolint: unparam
```
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply internal/resources/packages/packages_mock_test.go]: updated

File: examples/data-sources/package/data-source.tf [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Read Tanzu Mission Control package : fetch cluster package details
+data "tanzu-mission-control_package" "get_cluster_package" {
+  name           = "test-package-version" # Required
+
+  metadata_name  = "package-metadata-name" # Required
+
+  scope {
+    cluster {
+      cluster_name            = "testcluster" # Required
```
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply examples/data-sources/package/data-source.tf]: updatd

File: internal/models/package/cluster/package_spec.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+	return nil
+}
+
+// swagger:model k8s.io.apimachinery.pkg.runtime.RawExtension
+type K8sIoApimachineryPkgRuntimeRawExtension struct {
+
+	// Raw is the underlying serialization of this object.
+	//
+	// TODO: Determine how to detect ContentType and ContentEncoding of 'Raw' data.
```
[the remark this answers is not in the document]
Comment (ishangupta-ds) [reply internal/models/package/cluster/package_spec.go:94]: Can you remove this TODO then or maybe make it a normal comment?

[the remark this answers is not in the document]
Comment (Shubbhang351) [reply internal/models/package/cluster/package_spec.go]: updated PTAL

**Review by ishangupta-ds — APPROVED** [review]

**Review by ramya-bangera — APPROVED** [review]

**Pull request merged**